ledger/src/blockstore.rs

```diff
@@ -3317,6 +3317,55 @@ impl Blockstore {
         );
         Ok(())
     }
+
+    /// Mark a root `slot` as connected, traverse `slot`'s children and update
+    /// the children's connected status if appropriate.
+    ///
+    /// A ledger with a full path of blocks from genesis to the latest root will
+    /// have all of the rooted blocks marked as connected such that new blocks
+    /// could also be connected. However, starting from some root (such as from
+    /// a snapshot) is a valid way to join a cluster. For this case, mark this
+    /// root as connected such that the node that joined midway through can
+    /// have their slots considered connected.
+    pub fn set_and_chain_connected_on_root(&self, slot: Slot) -> Result<()> {
+        info!(
+            "Marking slot {} and any full children slots as connected",
+            slot
+        );
+
+        let mut meta = self
```

Comment:
nit `let mut starting_meta`, so that it's more logically aligned with `child_meta`.

Reply:
Good call, I'll go with `root_meta` to align with your suggestion to change the parameter name to `root`